File: app/src/main/java/com/example/spotibae/Activities/Welcome/WelcomeScreen.java
package com.example.spotibae.Activities.Welcome;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import com.example.spotibae.R;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class WelcomeScreen extends AppCompatActivity {

    private FirebaseAuth mAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        // Initialize Firebase Auth
        mAuth = FirebaseAuth.getInstance();
        this.setListeners();
    }

    @Override
    public void onStart() {
        super.onStart();
        // Check if user is signed in (non-null) and update UI accordingly.
        FirebaseUser currentUser = mAuth.getCurrentUser();
        if(currentUser != null){
            skipToHome();
        }
    }

    private void setListeners() {
        Button loginButton = findViewById(R.id.loginButton);
        Button signUpButton = findViewById(R.id.signUpButton);
        loginButton.setOnClickListener(v -> login());
        signUpButton.setOnClickListener(v -> signUp());
    }

    private void login() {
        Intent intent = new Intent(this, Login.class);
        this.startActivity(intent);
    }

    private void signUp() {
        Intent intent = new Intent(this, Signup.class);
        this.startActivity(intent);
    }

    private void skipToHome() {
        Intent intent = new Intent(this, BaseActivity.class);
        this.startActivity(intent);
    }

}

File: app/src/main/java/com/example/spotibae/Activities/Welcome/WelcomeScreen.kt
package com.example.spotibae.Activities.Welcome

import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import android.os.Bundle
import com.example.spotibae.R
import android.content.Intent
import android.view.View
import android.widget.Button

class WelcomeScreen : AppCompatActivity() {
    private var mAuth: FirebaseAuth? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        // Initialize Firebase Auth
        mAuth = FirebaseAuth.getInstance()
        setListeners()
    }

    public override fun onStart() {
        super.onStart()
        // Check if user is signed in (non-null) and update UI accordingly.
        val currentUser = mAuth!!.currentUser
        if (currentUser != null) {
            skipToHome()
        }
    }

    private fun setListeners() {
        val loginButton = findViewById<Button>(R.id.loginButton)
        val signUpButton = findViewById<Button>(R.id.signUpButton)
        loginButton.setOnClickListener { login() }
        signUpButton.setOnClickListener { signUp() }
    }

    private fun login() {
        val intent = Intent(this, Login::class.java)
        this.startActivity(intent)
    }

    private fun signUp() {
        val intent = Intent(this, Signup::class.java)
        this.startActivity(intent)
    }

    private fun skipToHome() {
        val intent = Intent(this, BaseActivity::class.java)
        this.startActivity(intent)
    }
}